friends/src/App.js

```diff
@@ -0,0 +1,196 @@
+import React, { Component } from 'react';
+import './App.css';
+import axios from 'axios';
+import { Route, Link } from 'react-router-dom';
+
+import Friends from './components/friends';
+import AddFriend from './components/addFriend';
+
+// server url
+let url = 'http://localhost:5000/friends';
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state= {
+      data: [],
+      inputName: '',
+      inputAge: '',
+      inputEmail: '',
+    }
+  }
+
+  // request data from server and setState with response
+  componentDidMount() {
+    axios
+      .get(url)
+      .then(res => {
+        this.setState({
+          data: res.data,
+        });
+      })
+      .catch(err => {
+        console.log(err);
+      })
+  }
+
+  // updates state keys (inputName, inputAge, inputEmail) as user is typing in input fields; invoked with onChange
+  // passed as props to ADDFRIEND component
+  handleChange = e => {
+    this.setState({
+      [e.target.name]: e.target.value
+    });
+  }
+
+  // used for the clear button to reset our state keys (inputName, inputAge, inputEmail) and in turn clears our input fields
+  // passed as props to ADDFRIEND component
+  clearFields = e => {
+    e.preventDefault();
+    this.setState({
+      inputName: "",
+      inputAge: "",
+      inputEmail: "",
+    });
+  }
+
+  // invoked when a new friend object is to be created, sends request to server to create new object and clears state keys (inputName, inputAge, inputEmail)
+  // passed as props to ADDFRIEND component
+  addFriend = (name, age, email) => {
+    // we want to check if all the input fields are populated befoe sending server request
+    if(this.state.inputName && this.state.inputAge && this.state.inputEmail){ 
+      axios
+        .post(url, {name, age , email})
+        .then(res => {
+          console.log(res)
+          this.setState({
+            data: res.data,
```

Comment:
Why not use [object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring) to make this a little cleaner?
```javascript
.then(({ data }) => {
  this.setState({
    data,
    inputName: '',
    // ...
```